src/fragments/start/getting-started/reactnative/integrate.mdx

```diff
@@ -0,0 +1,408 @@
+
+Inside our `useEffect()` hook, we will add a `subscription` to data updates.
+
+```js
+  useEffect(() => {
+
+    //function to query the todolist and store the todolist in state. 
+    const onQuery = async () => {
+      const todos_list = await DataStore.query(Todo);
+      setTodos(todos_list);
+    };
+
+    //load todos on first render 
+    onQuery();
+
+    //subscribe to data updates and fetch the todos using onQuery().
+    //msg variable is the updated todo item and can be used for further customizations.
+    const subscription = DataStore.observe(Todo).subscribe((msg) => {
+      //onQuery will be called each time a data update subscription is received.
+      //onQuery will update the todos list with the latest data/
+      onQuery();
+    });
+
+    //unsubscribe to data updates so that we don’t introduce a memory leak.
+    return function cleanup() {
+      subscription.unsubscribe();
+    }
+  }, []);
+```
+
+If you restart your app now, you should see that newly added todos will also start showing up on the list. The items look like they can be check off and marked as completed, but when pressed, they don’t seem to do anything right now. Let’s learn how to update existing data.
+
+### Updating a Todo
+
+Updating an existing data entry looks a lot like creating a new one. It’s important to note, however, that models in DataStore are *immutable*. So, to update a record you must use a model’s `copyOf` function rather than manipulating its properties directly.
+
+Update the `setComplete` function in the `TodoList` component.
+
+```js
+  async function setComplete(updateValue, todo) {
+    //update the todo item with updateValue
+    await DataStore.save(
+      Todo.copyOf(todo, updated => {
+        updated.isComplete = updateValue
+      })
+    );
+  }
+```
+
+That’s it! Restart your app and you should now be able to toggle a todo between completed and not completed states. We’re almost done here but what if you want to delete an item from your todo list? We’ll go over how to do that next.
+
+### Deleting a Todo
+
+Deleting an existing data entry is even easier than updating one since you don’t need to copy the instance to delete it. It may not be the best user experience to do this but, for this app, we will trigger deletion of a todo item to a long press of the item.
+
+Update the `deleteTodo()` function in the `TodoList` component.
+
+```js
+  async function deleteTodo(todo) {
+    try {
+      await DataStore.delete(todo);
+    } catch (e) {
+      console.log('Delete failed: $e');
+    }
+  }
+```
+
+Reload your app once more and you should now be able to long press an item to delete it. Now we have a fully featured CRUD application that saves and retrieves data on the local device, which means this app **works without an AWS account or even an internet connection.** Next, we’ll connect it to AWS and make sure the data is available in the cloud.
```

Comment:
At the end of this section - can we add the full file with all the functions added to it - so that developers can copy and use all of the functionality if they need to?

Reply:
Abdallah, I believe that would make the integrate section too long and make it seem harder to our users. We already have a link to sample Github repo at the start of the section. Maybe, we can add a link for the sample Home.js file from the same github repo. What do you think?

Reply:
Yes! I like that a lot, @chintannp! Thank you for the suggestion!

Reply:
Cool. I have added the link to the complete `Home.js` file.